src/object/obj_tx.c

```diff
@@ -642,7 +652,8 @@ dc_tx_check(daos_handle_t th, bool check_write, struct dc_tx **ptx)
 				"DTX is not full enabled.\n");
 			D_GOTO(out, rc = -DER_NO_PERM);
 		}
-	} else if (tx->tx_status != TX_OPEN) {
+	} else if (tx->tx_status != TX_OPEN &&
+		   tx->tx_status != TX_COMMITTED) {
```

Comment:
hmm why do we allow fetch from an already committed TX?
feels weird to me that this is allowed

Reply:
maybe we discussed & agreed to this on one of the calls and i don't recall.. but just wanted to make sure that we said we should allow this.

Reply:
> hmm why do we allow fetch from an already committed TX?
> feels weird to me that this is allowed

Why not allow that? I cannot remember we have some agreement for that before, sorry.
If not allow that, then when the user wants to verify its data after commit, how can he/she do?